servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -2957,45 +2957,45 @@ void RendererSceneRenderRD::_setup_lights(const PagedArray<RID> &p_lights, const
 					continue;
 				}
 
-				const real_t distance = camera_plane.distance_to(li->transform.origin);
+				const real_t depth = camera_plane.distance_to(li->transform.origin);
 
 				if (light_storage->light_is_distance_fade_enabled(li->light)) {
+					// Use `distance_squared_to()` to speed up distance checks.
+					const real_t distance = p_camera_transform.origin.distance_squared_to(li->transform.origin);
 					const float fade_begin = light_storage->light_get_distance_fade_begin(li->light);
 					const float fade_length = light_storage->light_get_distance_fade_length(li->light);
 
-					if (distance > fade_begin) {
-						if (distance > fade_begin + fade_length) {
-							// Out of range, don't draw this light to improve performance.
-							continue;
-						}
+					if (distance > Math::pow(fade_begin + fade_length, 2)) {
```

Comment:
Its probably better to do a multiply instead of a ``pow()`` call. Looking around online it sounds like GCC with -O3 optimizes pow(x, 2) to about the same as x*x but many other combinations of compiler/settings don't